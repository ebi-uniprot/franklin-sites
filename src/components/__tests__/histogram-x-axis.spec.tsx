import { render, screen } from '@testing-library/react';
import XAxis from '../histogram-x-axis';

describe('XAxis component', () => {
  it('should render with correct tickmarks', () => {
    const { asFragment } = render(
      <XAxis min={0} max={10} interval={5} yPos={0} width={100} />
    );
    expect(asFragment()).toMatchSnapshot();
    const labels = screen
      .getAllByText(/\d+/)
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      .map(({ __data__: label }) => label);
    expect(labels).toEqual([0, 5, 10]);
  });
});
