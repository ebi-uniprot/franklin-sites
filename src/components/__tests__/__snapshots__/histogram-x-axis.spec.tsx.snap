// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XAxis component should render with correct tickmarks 1`] = `
<DocumentFragment>
  <svg
    height="80"
    style="top: 0px;"
    width="100"
  >
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
    >
      <path
        class="domain"
        d="M0.5,0.5H100.5"
        stroke="currentColor"
      />
      <g
        class="tick"
        opacity="1"
        transform="translate(0.5,0)"
      >
        <line
          stroke="currentColor"
          y2="0"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="6"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(50.5,0)"
      >
        <line
          stroke="currentColor"
          y2="0"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="6"
        >
          5
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(100.5,0)"
      >
        <line
          stroke="currentColor"
          y2="0"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="6"
        >
          10
        </text>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;
